File: benchmarks/computing/matrix-multiplication.js
const matrixMultiplication = (matrix1, matrix2) => {
  const result = new Array(matrix1.length).fill(0).map(() => new Array(matrix2[0].length).fill(0))

  for (let i = 0; i < matrix1.length; i++) {
    for (let j = 0; j < matrix2[0].length; j++) {
      for (let k = 0; k < matrix2.length; k++) {
        result[i][j] += matrix1[i][k] * matrix2[k][j]
      }
    }
  }

  return result
}

function createEmptyMatrix(rows, cols) {
  let result = [];
  for (let i = 0; i < rows; i++) {
    let row = [];
    for (let j = 0; j < cols; j++) {
      row.push(0);
    }
    result.push(row);
  }
  return result;
}

function generateRandomMatrix(matrixSize) {
  const matrixA = new Array(matrixSize);

  for (let i = 0; i < matrixSize; i++) {
    matrixA[i] = new Array(matrixSize);

    for (let j = 0; j < matrixSize; j++) {
      matrixA[i][j] = Math.random();
    }
  }

  return matrixA;
}

const matrixSize = 700 // Adjust the matrix size as needed
const matrixA = generateRandomMatrix(matrixSize)
const matrixB = generateRandomMatrix(matrixSize)

// const matrixA = []
// const matrixB = []

// for (let i = 0; i < matrixSize; i++) {
//   matrixA.push(Array.from({ length: matrixSize }, () => Math.random()))
//   matrixB.push(Array.from({ length: matrixSize }, () => Math.random()))
// }

matrixMultiplication(matrixA, matrixB)

File: benchmarks/computing/matrix-multiplication/matrix-multiplication.go
package main

import (
	"math/rand"
	"time"
)

func matrixMultiplication(matrix1, matrix2 [][]float64) [][]float64 {
	rows1 := len(matrix1)
	cols1 := len(matrix1[0])
	cols2 := len(matrix2[0])

	result := make([][]float64, rows1)
	for i := 0; i < rows1; i++ {
		result[i] = make([]float64, cols2)
	}

	for i := 0; i < rows1; i++ {
		for j := 0; j < cols2; j++ {
			for k := 0; k < cols1; k++ {
				result[i][j] += matrix1[i][k] * matrix2[k][j]
			}
		}
	}

	return result
}

func createEmptyMatrix(rows, cols int) [][]float64 {
	result := make([][]float64, rows)
	for i := 0; i < rows; i++ {
		result[i] = make([]float64, cols)
	}
	return result
}

func generateRandomMatrix(matrixSize int) [][]float64 {
	randSource := rand.NewSource(time.Now().UnixNano())
	rand := rand.New(randSource)
	matrix := createEmptyMatrix(matrixSize, matrixSize)
	for i := 0; i < matrixSize; i++ {
		for j := 0; j < matrixSize; j++ {
			matrix[i][j] = rand.Float64()
		}
	}
	return matrix
}

func main() {
	matrixSize := 700 // Adjust the matrix size as needed
	matrixA := generateRandomMatrix(matrixSize)
	matrixB := generateRandomMatrix(matrixSize)

	matrixMultiplication(matrixA, matrixB)
}
